File: 프로그래머스/Lv.1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
# 프로그래머스 42576 완주하지 못한 선수 LV.1
def solution(participant, completion):
    participant.sort()
    completion.sort()
    for i in range(len(completion)):
        if participant[i] != completion[i]:
            return participant[i]
    return participant[-1]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> map = new HashMap<>();
        for (String p: participant) {
            map.put(p, map.getOrDefault(p, 0) + 1);
        }
        for (String c: completion) {
            if (map.containsKey(c)) {
                map.put(c, map.get(c) - 1);
            }
            if (map.get(c) == 0) {
                map.remove(c);
            }
        }

        for (String x: map.keySet()) {
            return x;
        }
        return "";
    }
}